### PR TITLE
Feature/ab 1001 speedup trufflehog semgrep v2

### PR DESCRIPTION
**Pull request by Anthony-Mckale**
- updated ezerc.toml to fit new format
- deleted old demo and moved them to eze-goats
  
AUTO CONFIG
- added custom add custom auto config json to eze test
- deleting old language based auto .ezerc.toml engine
  from eze.core.language import LanguageManager
- built new one around eze/data/default_autoconfig.json (details in README.md)
  from eze.core.autoconfig import AutoConfigRunner

SEMGREP
- speeded up semgrep by 80% in windows using -e "WINDOWS_DOCKER_WORKAROUND=true" details in README.md
- semgrep now auto detects files and which rules to apply

TRUFFLEHOG
- speeded up trufflehog 80% by adding gitignores to exclude list
- truffle now has a default exclude list
   from eze.utils.file_scanner import IGNORED_FOLDERS

[AB#1001](https://dev.azure.com/riversafe/0b0abc1f-f46a-4d96-a470-e1c546e3cf29/_workitems/edit/1001)

### CODE REVIEW

**Review by mgonzales-rs — COMMENTED** [review]
testing

File: README-DEVELOPMENT.md [hunk 1]
```diff
@@ -164,6 +164,46 @@ eze test
...
+  },
+  "tools": {
+    "<tool-id>": {
+      "_help_message": "<DEVELOPER COMMENTS>",
+      "enabled_always": "true or false",
+      "enable_on_file": [
+        "<LIST OF FILE NAMES IN FOUND IN CODEBASE WILL ENABLE>"
+      ],
+      "enable_on_file_ext": [
+        "<LIST OF FILE NAMES IN FOUND IN CODEBASE WILL ENABLE>"
```
Comment (mgonzales-rs) [inline README-DEVELOPMENT.md:190]: LIST OF FILE EXTENSIONS ...

File: eze/cli/commands/test_commands.py [hunk 2]
```diff
@@ -77,7 +78,7 @@ def test_online_command(state, config_file: str, url: str) -> None:
 @click.option(
     "--url",
     "-u",
-    help="Specify the url of the remote repository to run scan. ex https://user:pass@github.com/repo-url",
+    help="Specify the url of the remote repository to run scan. ex https://user:pass@github.com/repo-url",  # nosecret
```
Comment (mgonzales-rs) [inline eze/cli/commands/test_commands.py:81]: quick qs, what is this for (the comment), for false positive in eze test?

File: eze/cli/commands/housekeeping_commands.py [hunk 3]
```diff
@@ -43,22 +43,22 @@ def housekeeping_group():
...
 
 
-@click.command("create-local-config", short_help="create local config file")
+@click.command("create-local-config", short_help="create local .ezerc.toml")
+@click.option("--autoconfig", type=click.Path(exists=True), help="File with custom autoconfig json", required=False)
 @debug_option
-def create_local_config_command():
-    """creates a default config for a user in their local location"""
+def create_local_config_command(autoconfig: click.Path = None):
+    """creates a automatically generated config for a user in their local location, based off whe"""
```
Comment (mgonzales-rs) [inline eze/cli/commands/housekeeping_commands.py:50]: creates an* automatically...   .  Quick qs: what does based off whe mean?

**Review by mgonzales-rs — APPROVED** [review]
looks good

File: README-DEVELOPMENT.md (hunk 1 above)
Comment (Anthony-Mckale) [reply README-DEVELOPMENT.md]: done 👍 )

File: eze/cli/commands/housekeeping_commands.py (hunk 3 above)
Comment (Anthony-Mckale) [reply eze/cli/commands/housekeeping_commands.py:50]: it's anthony being silly and not completing the sentence, silly dyslexia....

Comment (Anthony-Mckale) [reply eze/cli/commands/housekeeping_commands.py:50]: updated to : "creates a dynamically generated config for codebase in cwd, tools selected based off codebase contents"

thanks for the good spot

**Comment by mgonzales-rs** [comment]
just a suggestion to change one thing in eze-cli (that affects sarif reporter) 
image


according to Github actions startLine needs a minimum value of 0, so can we set the startLine to 1 when we have requirements.txt file?

**Comment by Anthony-Mckale** [comment]
> just a suggestion to change one thing in eze-cli (that affects sarif reporter) image
> 
> according to Github actions startLine needs a minimum value of 0, so can we set the startLine to 1 when we have requirements.txt file?

excellent suggestion, done

**Pull request merged**